### PR TITLE
EES-2597 - comments UI tests 

### PR DESCRIPTION
**Pull request by luke-h1**
image

### CODE REVIEW

**Comment by luke-h1** [comment]
CI failing due to out of date jest snapshot

File: tests/robot-tests/tests/admin/bau/comments.robot [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    user waits until element is visible    //*[@data-testid="Expand Details Section Add / View comments (1 unresolved)"]
+    user clicks element    //*[@data-testid="Expand Details Section Add / View comments (1 unresolved)"]
+    user scrolls down  400
+    user presses keys    I've fixed this issue    //*[@data-testid="comment-textarea"]
+    user clicks button    Add comment
+
+    # avoid set page view box intercepting clicks
+    user scrolls to element    //*[@data-testid="resolve-button"]
+    user waits until element is enabled    //*[@data-testid="resolve-button"]
+    user clicks button    Resolve
```
Comment (markhiveit) [inline tests/robot-tests/tests/admin/bau/comments.robot:177]: At this point there are two `Resolve` buttons. Maybe we could do something like this?
```
${comment1}=  user gets comment   Comment text here
user clicks button  Resolve   ${comment1}
```

Where `user gets comment` grabs the `li` that contains that comment-content

Not sure if this is worth the effort though - so could just leave this.

**Review by markhiveit — CHANGES_REQUESTED** [review]

File: tests/robot-tests/tests/admin/bau/comments.robot [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    user clicks link    Edit this release
+
+Navigate to content section as bau1
+    [Tags]    HappyPath
+    user clicks link    Content
+    user closes Set Page View box
+
+Add review comment for first content block
+    [Tags]    HappyPath
+    user scrolls to element    //*[@data-testid="accordionSection"]
```
[the remark this answers is not in the document]
Comment (luke-h1) [reply tests/robot-tests/tests/admin/bau/comments.robot:127]: Nice! Completely forgot about this custom selector

File: src/explore-education-statistics-admin/src/components/editable/Comments.tsx [hunk 3]
```diff
@@ -217,7 +218,7 @@ const Comments = ({
               <li key={comment.id}>
                 {commentDetail}
 
-                <p>{comment.content}</p>
+                <p data-testid="comment-content">{comment.content}</p>
```
Comment (markhiveit) [inline src/explore-education-statistics-admin/src/components/editable/Comments.tsx:221]: I would add the same data-testid on to line 201 too, so we can use it to select the content of resolved comments as well.

**Review by markhiveit — CHANGES_REQUESTED** [review]

File: tests/robot-tests/tests/admin/bau/comments.robot [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    ...    //*[@data-testid="Expand Details Section Add / View comments (1 unresolved)"]    10
+
+    ${comment_2}=    user gets comment    Fixed the problem
+    user clicks button    Resolve    ${comment_2}
+
+Move publication into 'Higher Review'
+    [Tags]    HappyPath
+    user clicks link    Sign off
+    user clicks button    Edit release status
+    user clicks element    //*[@id="releaseStatusForm-approvalStatus-HigherLevelReview"]
```
Comment (markhiveit) [inline tests/robot-tests/tests/admin/bau/comments.robot:188]: Can use `id:releaseStatusForm-approvalStatus-HigherLevelReview` here.

File: tests/robot-tests/tests/libs/common.robot [hunk 5]
```diff
@@ -760,3 +760,7 @@ lookup or return webelement
         ${element}=    get child element    ${parent}    ${selector_or_webelement}
     END
     [Return]    ${element}
+
+user closes Set Page View box
+    user clicks element    //*[@id="pageViewToggleButton"]
+    user waits until element is not visible    //*[@id="editingMode"]
```
Comment (markhiveit) [inline tests/robot-tests/tests/libs/common.robot:766]: Can use the `id:` locator on these two lines.

File: tests/robot-tests/tests/admin/bau/comments.robot [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+Navigate to content section as bau1
+    [Tags]    HappyPath
+    user clicks link    Content
+    user closes Set Page View box
+
+Add review comment for first content block
+    [Tags]    HappyPath
+    user scrolls to element    testid:accordionSection
+    user clicks element    xpath://button[text()="First content section"]
```
Comment (markhiveit) [inline tests/robot-tests/tests/admin/bau/comments.robot:129]: Can use `user clicks button  First content section` here.

File: tests/robot-tests/tests/admin/bau/comments.robot [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+Navigate to content section to resolve comments
+    [Tags]    HappyPath
+    user clicks link    Content
+    user waits until h1 is visible    ${PUBLICATION_NAME}
+
+Resolve comments
+    [Tags]    HappyPath
+    user closes Set Page View box
+    user scrolls to element    testid:accordionSection
+    user clicks element    xpath://button[text()="First content section"]
```
Comment (markhiveit) [inline tests/robot-tests/tests/admin/bau/comments.robot:163]: Can use `user clicks button    First content section` here.

File: tests/robot-tests/tests/admin/bau/comments.robot [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+    user clicks element    //*[@data-testid="Expand Details Section Add / View comments (1 unresolved)"]
```
Comment (markhiveit) [inline tests/robot-tests/tests/admin/bau/comments.robot:164]: Is the `testid:`  custom locator working now in all instances? If so can use `testid:Expand Details Section Add / View comments (1 unresolved)` here. There are a couple of other instances of data-testids being selected by xpath that could potentially use `testid:` too.

**Review by markhiveit — CHANGES_REQUESTED** [review]

File: tests/robot-tests/tests/libs/common.robot (hunk 5 above)
Comment (luke-h1) [reply tests/robot-tests/tests/libs/common.robot:766]: Cool, didn't realise this was available

**Review by markhiveit — CHANGES_REQUESTED** [review]

**Review by markhiveit — APPROVED** [review]

**Pull request merged**